runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -156,6 +154,172 @@ static TR::Node *lowerCASValues(
+ *  |treetop               |
+ *  |  icall acmpHelper    |
+ *  |    aload lhs         |
+ *  |    aload rhs         |
+ *  |ificmpeq --> ...      |
+ *  |  ==> icall           |
+ *  |  iconst 0            |
+ *  |BBEnd                 |
+ *  +----------------------+
+ *
+ *  ...becomes...
+ *
+ *  +----------------------+
+ *  |ttprev                |
+ *  |iRegStore x           |
+ *  |  iconst 1            |
+ *  |ifacmpeq  -->---------*---------+
+ *  |  aload lhs           |         |
+ *  |  aload rhs           |         |
+ *  |BBEnd                 |         |
+ *  +----------------------+         |
+ *  |BBStart (extension)   |         |
+ *  |iRegStore x           |         |
+ *  |  icall acmpHelper    |         |
+ *  |    aload lhs         |         |
+ *  |    aload rhs         |         |
+ *  |BBEnd                 |         |
+ *  +----------------------+         |
+ *        |                          |
+ *        +--------------------------+
+ *        |
+ *        v
+ *  +-----------------+
+ *  |BBStart
+ *  |ificmpeq --> ... |
+ *  |  iRegLoad x     |
+ *  |  iconst 0       |
+ *  |BBEnd            |
+ *  +-----------------+
+ *
+ */
+void
+J9::CodeGenerator::fastpathAcmpHelper(TR::Node *node, TR::TreeTop *tt)
+   {
+   TR::Compilation* comp = self()->comp();
+   traceMsg(comp, "<LEO> Detected call to substitutabilityComparisonSymbol in n%dn\n", node->getGlobalIndex());
+
+   TR::CFG* cfg = comp->getFlowGraph();
+   cfg->setStructure(0);
+
+   // anchor call node after split point to ensure the returned value goes into
+   // either a temp or a global register
+   auto* anchoredCallTT = TR::TreeTop::create(comp, tt, TR::Node::create(TR::treetop, 1, node));
+   traceMsg(comp, "<LEO> Anchoring call node under treetop n%dn (0x%p)\n", anchoredCallTT->getNode()->getGlobalIndex(), anchoredCallTT->getNode());
+
+   // anchor the call arguments just before the call
+   // this ensures the values are live before the call so that we can
+   // propagate their values in global registers if needed
+   auto* anchoredCallArg1TT = TR::TreeTop::create(comp, tt->getPrevTreeTop(), TR::Node::create(TR::treetop, 1, node->getFirstChild()));
+   auto* anchoredCallArg2TT = TR::TreeTop::create(comp, tt->getPrevTreeTop(), TR::Node::create(TR::treetop, 1, node->getSecondChild()));
+   traceMsg(comp, "<LEO> Anchoring call argument n%dn and n%dn under treetops n%dn and n%dn\n",
+      node->getFirstChild()->getGlobalIndex(), node->getSecondChild()->getGlobalIndex(), anchoredCallArg1TT->getNode()->getGlobalIndex(), anchoredCallArg2TT->getNode()->getGlobalIndex());
+
+   // put non-helper call in its own block by block splitting at the
+   // next treetop and then at the current one
+   TR::Block* prevBlock = tt->getEnclosingBlock();
+   comp->dumpMethodTrees("Initial trees");
+   TR::Block* targetBlock = prevBlock->splitPostGRA(tt->getNextTreeTop(), cfg, true, NULL);
+   TR::Block* callBlock = prevBlock->split(tt, cfg, true, NULL);
```

Comment:
@Leonardo2718  I have just looked into the part where you are splitting blocks. When you create a `callBlock`, you do not need to set `fixupCommoning` to `true`, this will cause it to walk the tree-top before the split point (Although, there might not be nothing to common). Another reason for not to set up `fixupCommoning` to `true` if you are setting the callBlock as extension of previous block.